server/fleet/microsoft_mdm.go

```diff
@@ -224,7 +224,7 @@ func (req *SoapRequest) IsValidRequestSecurityTokenMsg() error {
 	}
 
 	reqEnrollType, err := req.Body.RequestSecurityToken.GetContextItem(mdm.ReqSecTokenContextItemEnrollmentType)
-	if err != nil || reqEnrollType != mdm.ReqSecTokenEnrollType {
```

Comment:
Is there a defined range of types this could be in addition to `mdm.ReqSecTokenEnrollType`? How much effort would it be to enumerate the additional allowed types here?

Reply:
I initially thought that this was going to be the only value, but after a second look at the spec I found that there are 3 enroll type values documented in the MS-MDE2 spec [here](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-mde2/f7553554-b6e1-4a0d-abd6-6a2534503af7). See `ProviderID/EnrollmentType` for more details. "Device" is the enrollment type used during the MDM enrollment happening during the OOBE, then "Full" is the enrollment type sent in a post-OOBE scenario, and "AppManagement" is the enrollment type used by the MAM enrollment (we are not supporting this fort now). Sp I've added a new check to whitelist the enrollment scenarios we support